modules/phase_field/src/utils/Euler2RGB.C

```diff
@@ -289,28 +289,36 @@ euler2RGB(unsigned int sd, Real phi1, Real PHI, Real phi2, unsigned int phase, u
       // Increment to next symmetry operator if not in SST
       else
         index++;
+
+      // Error check if no solution found
+      if (index == nsym)
+        mooseError("Euler2RGB failed to map the supplied Euler angle into the SST!");
     }
 
     //  Adjust maximum chi value to ensure it falls within the SST (cubic materials only)
     if (sym == 43)
-      chi_max2 = std::acos(std::sqrt(1.0 / (2.0 + (std::tan(Utility::pow<2>(eta))))));
+      chi_max2 = std::acos(std::sqrt(1.0 / (2.0 + (Utility::pow<2>(std::tan(eta))))));
     else
       chi_max2 = pi / 2;
 
-    //  Calculate the RGB color values and make adjustments to maximize colorspace
+    // Calculate the RGB color values and make adjustments to maximize colorspace
     red = std::abs(1.0 - (chi / chi_max2));
     blue = std::abs((eta - eta_min) / (eta_max - eta_min));
     green = 1.0 - blue;
 
     blue = blue * (chi / chi_max2);
     green = green * (chi / chi_max2);
 
+    // Error check to prevent negative values being input into the square root calculation
+    if (red < 0 || green < 0 || blue < 0)
+      mooseError("RGB componenet values cannot be negative!");
```

Comment:
Again, do we need a hard fail here? (also typo `componenet`)

Reply:
`red` can never be negative! For the other two I'd have to look closer at all those acos arguments, but this may be a case for an assertion instead.

Reply:
I guess this is kind of redundant.  If the Euler angles map into the SST correctly, these will never be negative. I can remove it or change is to a warning if you prefer.

Reply:
change to a `mooseAssert`